modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterManager.java

```diff
@@ -1660,9 +1718,31 @@ String getMetadataValue(String name) {
 
         // check if heterogeneous CRS
         if (name.equalsIgnoreCase(AbstractGridCoverage2DReader.MULTICRS_READER)) {
-            return String.valueOf(configuration.getCatalogConfigurationBean().isHeterogeneousCRS());
+            return String.valueOf(heterogeneousCRS);
         }
 
+        if (name.equalsIgnoreCase(AbstractGridCoverage2DReader.MULTICRS_EPSGCODES)
+                && heterogeneousCRS) {
+
+            // Extract the internal EPSG Codes found on the catalog
+            String crsAttribute = null;
+            try {
+                crsAttribute = getCrsAttribute();
+                if (crsAttribute != null) {
+                    Set<String> crsSet = extractDomain(crsAttribute);
+                    for (String crs : crsSet) {
+                        // Cache the results for future queries
```

Comment:
This seems like opportunistic caching, as in "while we computed the list of epsg codes supported, let's add them in the cache for the code by code checks", but I'm just guessing here. Would be best if it was clear.

Reply:
K. Improving that comment